### PR TITLE
Arachnid zombie fix

### PR DESCRIPTION
**Pull request by PixelTheKermit**
## About the PR

Arachnids no longer get hands when becoming a zombie.
image

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- fix: Arachnid zombies no longer get functional hands.

### CODE REVIEW

**Comment by PixelTheKermit** [comment]
I'm not sure if test failures are my fault, yell at me if so.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

File: Content.Shared/Hands/EntitySystems/SharedHandsSystem.cs [hunk 1]
```diff
@@ -77,6 +77,35 @@ public virtual void RemoveHand(EntityUid uid, string handName, HandsComponent? h
...
+    }
+
+    private void RemoveHands(EntityUid uid, IEnumerable<Hand> hands, HandsComponent handsComp)
+    {
+        if (!hands.Any())
+            return;
+
+        var hand = hands.First();
+        SetActiveHand(uid, hand, handsComp);
+        DoDrop(uid, hand, handsComp: handsComp);
```
Comment (EmoGarbage404) [inline Content.Shared/Hands/EntitySystems/SharedHandsSystem.cs:102]:
```suggestion
```

this is apparently handled by the RemoveHand call
```cs
public virtual void RemoveHand(EntityUid uid, string handName, HandsComponent? handsComp = null)
{
    if (!Resolve(uid, ref handsComp, false))
        return;

    if (!handsComp.Hands.Remove(handName, out var hand))
        return;

    TryDrop(uid, hand, null, false, true, handsComp);
    hand.Container?.Shutdown();
    handsComp.SortedHands.Remove(hand.Name);

    if (handsComp.ActiveHand == hand)
        TrySetActiveHand(uid, handsComp.SortedHands.FirstOrDefault(), handsComp);

    RaiseLocalEvent(uid, new HandCountChangedEvent(uid), false);
    Dirty(handsComp);
}
```

Comment (PixelTheKermit) [reply Content.Shared/Hands/EntitySystems/SharedHandsSystem.cs:102]: Alright give me like half an hour and I'll do it

File: Content.Shared/Hands/EntitySystems/SharedHandsSystem.cs [hunk 2]
```diff
@@ -77,6 +77,35 @@ public virtual void RemoveHand(EntityUid uid, string handName, HandsComponent? h
...
+    /// <param name="uid"></param>
+    /// <param name="handsComp"></param>
+
+    public void RemoveHands(EntityUid uid, HandsComponent? handsComp = null)
+    {
+        if (!Resolve(uid, ref handsComp))
+            return;
+
+        RemoveHands(uid, EnumerateHands(uid), handsComp);
+        RemComp(uid, handsComp);
```
Comment (EmoGarbage404) [inline Content.Shared/Hands/EntitySystems/SharedHandsSystem.cs:92]: this should still be in zombie code. removing all of the hands shouldn't necessarily mean removing the component as well.

Comment (PixelTheKermit) [reply Content.Shared/Hands/EntitySystems/SharedHandsSystem.cs:92]: Noted

**Pull request merged**